### PR TITLE
Update documentation of wrappedValue force unwrap

### PR DESCRIPTION
**Pull request by rogerluan**
Hey 👋 

I believe it makes sense to inform users of the lib about this caveat, so I moved it from private to the public instance, and updated the copy a bit. Lemme know your thoughts!

### CODE REVIEW

**Comment by rogerluan** [comment]
On a second thought, I think we'd better throw like this @JARMourato :

```diff
- return _value as! U
+ if value == nil {
+     preconditionFailure(message: "Explain why this cant be done, add the docs here")
+ } else {
+     return _value as! U
+ }
```

Thoughts? 😊

**Comment by JARMourato** [comment]
[This](https://github.com/apple/swift/blob/c4c27f63048826678d7a7e6cde4b5797d291cec6/test/decl/var/property_wrappers.swift#L115) is the error you'd get from the compiler if one were to try to make the initializer throw. It is a known limitation of property wrappers, at least until now. 

This is the only case where it would crash

```Swift
struct Roger: Kodable {
    @Coding("rogdog") var name: String
}
let r = Roger()
print(r.name) // Crashes when getting the name property because the underlying `_value` is nil
```

I think it is okay to leave it like this, at least for the use I give to the lib. 

I like your suggestion. However, I'd prefer if you use `fatalError` instead of `preconditionFailure`.

**Comment by rogerluan** [comment]
Seems good to me! 
The crash seems fine (like you said, given the context of the lib), but giving a better user feedback is always important - and `fatalError` does make more sense to me as well 👌

File: Sources/Kodable/KodableTransformable.swift [hunk 1]
```diff
@@ -22,12 +22,12 @@ public protocol KodableTransform {
     public typealias OriginalType = T.From
     public typealias TargetType = T.To
 
-    // Note: this might crash if an instance of a type uses the property wrapper with a non-optional type
-    /// and it is not decoded. (i.e. the `_value` is not initialized a this it is nil.)
     private func _wrappedValue<U>(_: U.Type) -> U {
         _value as! U
```
Comment (rogerluan) [inline Sources/Kodable/KodableTransformable.swift]:
```suggestion
        if _value == nil {
            fatalError("Trying to access a non optional property that has not been decoded - the property value is nil internally")
        } else {
            _value as! U
        }
```

Just missing the property key as idk how to get it within this context @JARMourato 🙈

**Review by JARMourato — APPROVED** [review]

File: Sources/Kodable/KodableTransformable.swift [hunk 2]
```diff
@@ -22,12 +22,16 @@ public protocol KodableTransform {
...
     public typealias TargetType = T.To
 
-    // Note: this might crash if an instance of a type uses the property wrapper with a non-optional type
-    /// and it is not decoded. (i.e. the `_value` is not initialized a this it is nil.)
     private func _wrappedValue<U>(_: U.Type) -> U {
-        _value as! U
+        if _value == nil {
+            fatalError("Trying to access a non optional property that has not been decoded - the property value is nil internally")
+        } else {
+            _value as! U
```
Comment (JARMourato) [inline Sources/Kodable/KodableTransformable.swift]:
```suggestion
           return _value as! U
```

**Review by JARMourato — CHANGES_REQUESTED** [review]

**Review by JARMourato — APPROVED** [review]

File: Sources/Kodable/KodableTransformable.swift [hunk 3]
```diff
@@ -22,12 +22,16 @@ public protocol KodableTransform {
...
 
-    // Note: this might crash if an instance of a type uses the property wrapper with a non-optional type
-    /// and it is not decoded. (i.e. the `_value` is not initialized a this it is nil.)
     private func _wrappedValue<U>(_: U.Type) -> U {
-        _value as! U
+        if _value == nil {
+            fatalError("Trying to access a non optional property that has not been decoded - the property value is nil internally")
+        } else {
+           return _value as! U
+        }
```
Comment (rogerluan) [inline Sources/Kodable/KodableTransformable.swift]:
```suggestion
        if _value == nil {
            fatalError("Trying to access a non optional property that has not been decoded - the property value is nil internally")
        } else {
            return _value as! U
        }
```

**Pull request merged**